packages/contentstack-branch/src/branch/diff.ts

```diff
@@ -0,0 +1,163 @@
+import chalk from "chalk";
+import forEach from "lodash/forEach";
+import startCase from "lodash/startCase";
+import camelCase from "lodash/camelCase";
+import { cliux } from "@contentstack/cli-utilities";
+import {
+  BranchOptions,
+  BranchDiffRes,
+  BranchDiffSummary,
+  BranchCompactTextRes,
+} from "../interfaces/index";
+import BranchDiffUtility from '../utils/diff';
+import {
+  askCompareBranch,
+  askBaseBranch,
+  askStackAPIKey,
+  selectModule
+} from "./interactive";
+
+
+export default class BranchDiff {
+  private options: BranchOptions;
+  public branchUtilityInstance: BranchDiffUtility;
+  public branchesDiffData: BranchDiffRes[];
+  public branchSummary: BranchDiffSummary;
+  public branchCompactTextRes: BranchCompactTextRes;
+
+  constructor(params: BranchOptions) {
+    this.options = params;
+  }
+
+  async run(): Promise<any> {
+    await this.validateMandatoryFlags();
+    await this.utilityInstance();
+    this.displaySummary();
+    this.displayBranchDiffTextAndVerbose();
+  }
+
+  /**
+   * @methods validateMandatoryFlags - validate flags and prompt to select required flags
+   * @returns {*} {Promise<void>}
+   * @memberof BranchDiff
+   */
+  async validateMandatoryFlags(): Promise<void> {
+    if (!this.options.baseBranch) {
+      this.options.baseBranch = await askBaseBranch();
```

Comment:
It should be provided using the config, if config not set, we should ask them to set the config by suggesting config command and exit the process